core/src/main/java/org/elasticsearch/search/query/QuerySearchResult.java

```diff
@@ -133,8 +149,17 @@ public void aggregations(InternalAggregations aggregations) {
      * Returns the profiled results for this search, or potentially null if result was empty
      * @return The profiled results, or null
      */
-    @Nullable public ProfileShardResult profileResults() {
-        return profileShardResults;
+    public ProfileShardResult consumeProfileResult() {
```

Comment:
maybe copy/paste the comment that you put on aggs that this API is useful to free up memory, it helped me understand why it was designed this way